src/expressions.js

```diff
@@ -40,3 +40,42 @@ exports.satisfiesExpression = function(scopeset, expr) {
     exports.satisfiesExpression(scopeset, subexpr)
   );
 };
+
+/**
+ * Given a scopeset and a scope-expression that it
+ * does not satisfy, explain why the scope-expression was
+ * not satisfied.
```

Comment:
Why is this a requirement... if it does satisfy the scope-expression then this _simplifies_ or _reduces_ to `{AllOf: []}`

Reply:
I suppose it is not a requirement. It is not enforced anywhere.

Reply:
or say that it results in `null` which is a pretty good reduction...